framework/aster-frame/src/lib.rs

```diff
@@ -69,6 +69,16 @@ pub fn init() {
     invoke_ffi_init_funcs();
 }
 
+/// Currently, since the filesystems are mounted during the OS initialization phase,
+/// it is necessary to have this API to enable interrupts for the proper functioning of block devices.
+/// If we move the step of mounting the filesystems to be done within the init process,
+/// this API can be removed.
+pub fn enable_interrupts() {
```

Comment:
First of all, this function should not be added the root module of `aster-frame`. There is the `trap` module, which is most related to IRQ. Move the function to this module. In fact, you can simple make the existing `arch::irq::enable_local()` function public. The proposed `enable_interrupts` and the existing `enable_local` are basically the same thing.

Second, the soundness of this function is questionable as exposing this function allows the Framework's users to enable interrupts when she should not. For example, she may write code.

```rust
let guard = lock.lock_irq_disabled();
enable_interrupts();
```

To say the least, exposing `enable_interrupts` breaks the assumption of `SpinLock::lock_irq_disabled()`. So it should be avoided.

So I suggest ading a FIXME.

```rust
// FIXME: this method may be unsound.
pub fn enable_interrupts() {
```

Third, the Rust doc seems to miss the point.

```rust
/// Currently, since the filesystems are mounted during the OS initialization phase,
/// it is necessary to have this API to enable interrupts for the proper functioning of block devices.
/// If we move the step of mounting the filesystems to be done within the init process,
/// this API can be removed.
```

The point is that _IRQs should be enabled as early as possible_. The quick fix in this PR, including adding this `enable_interrupts` method, suggests that IRQs are enabled as late as the time the CPU enters the user space for the first time. This is too late.

Reply:
I am not quite familiar with the code structure of `trap`, so I just add an API in lib.rs. 

I have moved the code into the irq.rs of trap module, and refine the expalin of this API.